src/backend/aspen/api/utils/phylo.py

```diff
@@ -54,13 +54,31 @@ def _rename_nodes_on_tree(
     """Given a tree, a mapping of identifiers to their replacements, rename the nodes on
     the tree.  If `save_key` is provided, then the original identifier is saved using
     that as the key."""
-    name = node["name"]
-    renamed_value = name_map.get(name, None)
+    gisaid_prefix = "hCoV-19/"
+
+    # The mixed situations we're dealing with here:
+    #  - The public identifiers in our database *sometimes* have gisaid prefixes on them
+    #  - The samples on a tree *sometimes* have gisaid prefixes on them.
+    #  - We want to match identifiers from trees and the db with the prefix *stripped*
+    #  - At the end of this method, we want *all* tree samples to have gisaid prefixes on them.
+    # So this means we have to:
+    #  - Strip gisaid prefixes from tree nodes before trying to match them to db samples
+    #  - Add gisaid prefixes to all public tree identifiers if they aren't already prefixed
+
+    # Strip the gisaid prefix from our tree identifier if it's present
+    tree_identifier = node["name"]
+    if tree_identifier.lower().startswith(gisaid_prefix.lower()):
+        tree_identifier = tree_identifier[len(gisaid_prefix) :]
+
+    renamed_value = name_map.get(tree_identifier, None)
+
+    # Make sure we have the gisaid prefix on this node when we output it.
+    node["name"] = f"{gisaid_prefix}{tree_identifier}"
```

Comment:
can we just add a check here to fix the NODE renaming issue? 

ex:  `node["name"] =f"{gisaid_prefix}{tree_identifier}" if not tree_identifier.startswith("NODE_") else tree_identifier`

Reply:
cool this is what i'm looking for. let me plug this in and test out 😉